esmvalcore/preprocessor/_other.py

```diff
@@ -85,3 +99,447 @@ def get_array_module(*args):
         if isinstance(arg, da.Array):
             return da
     return np
+
+
+def get_all_coords(
+    cube: Cube,
+    coords: Iterable[Coord] | Iterable[str] | None,
+) -> Iterable[Coord] | Iterable[str]:
+    """Get all desired coordinates in a cube."""
+    if coords is None:
+        coords = [c.name() for c in cube.dim_coords]
+        if len(coords) != cube.ndim:
+            raise ValueError(
+                f"If coords=None is specified, the cube "
+                f"{cube.summary(shorten=True)} must not have unnamed "
+                f"dimensions"
+            )
+    return coords
+
+
+def get_all_coord_dims(
+    cube: Cube,
+    coords: Iterable[Coord] | Iterable[str],
+) -> tuple[int, ...]:
+    """Get sorted list of all coordinate dimensions from coordinates."""
+    all_coord_dims = []
+    for coord in coords:
+        all_coord_dims.extend(cube.coord_dims(coord))
+    sorted_all_coord_dims = sorted(list(set(all_coord_dims)))
+    return tuple(sorted_all_coord_dims)
+
+
+def get_weights(
+    cube: Cube,
+    coords: Iterable[Coord] | Iterable[str],
+) -> np.ndarray | da.Array:
+    """Calculate suitable weights for given coordinates."""
+    npx = get_array_module(cube.core_data())
```

Comment:
this guy here `get_array_module` (and its use below, too) is causing a circular imports panic on good ole Python interpreter, via `_regrid` and `_other` (this module) - prob best bet would be to simply expand in code in your functions and not use the `get_array_module` function itself - though it's silly this way. If you asked me I'd put that function in `__init__` - it looks like an init-y function to me

Reply:
good call! Was thinking `_shared` when I mentioned init :beers: